index.js

```diff
@@ -172,3 +174,46 @@ form.addEventListener('submit', (event) => {
     showMessage(btnSubmit, ERROR);
   }
 });
+
+/* Preserve data in the browser */
+
+let contactFormData = {
+  data: [],
+  state: true,
+};
+
+const storeData = (item, fName) => {
+  let formData = {
+    name: '',
+    email: '',
+    message: '',
+  };
+
+  if (contactFormData.data.length > 0) {
+    [formData] = contactFormData.data;
+  }
+  formData[fName] = item.value;
+  contactFormData.data[0] = formData;
+  localStorage.setItem('contactFormData', JSON.stringify(contactFormData));
+};
+
+name.addEventListener('change', () => {
+  storeData(name, 'name');
+});
+
+email.addEventListener('change', () => {
+  storeData(email, 'email');
+});
+
+message.addEventListener('change', () => {
+  storeData(message, 'message');
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  contactFormData = JSON.parse(localStorage.getItem('contactFormData'));
+  if (contactFormData.data.length > 0) {
```

Comment:
- [ ] The data preservation is not working as you can see below 👇 Kindly fix the issues 👍 

![Animation](https://user-images.githubusercontent.com/50721479/209151719-2c52f296-10b5-4a94-aac1-22d884bbff17.gif)

This is local storage, it's not saving either
![image](https://user-images.githubusercontent.com/50721479/209151854-ad2b0f84-226f-4acf-8a0d-1c1d08d729e6.png)

And I have this error in console about this line (#214) 👇 
![image](https://user-images.githubusercontent.com/50721479/209151954-766528c4-4cef-4beb-990f-e1ccc62c3b7e.png)